### PR TITLE
Use viewport options for VNC console

### PR DESCRIPTION
**Pull request by skobyda**
There is an issue where if virtual machine system's resolution is set to something larger than VNC console element rendered in the browser, the view will be scaled down, but there will be an offset for mouse clicks:

[false.webm](https://user-images.githubusercontent.com/42733240/209139339-e08ce7fa-3fcb-45f5-a89b-e7a5e0ed103d.webm)

This should fix it:

[true-(2).webm](https://user-images.githubusercontent.com/42733240/209139322-ebb84f0a-7810-44f0-960f-df9d8bc0a1bc.webm)

Fixes https://github.com/cockpit-project/cockpit-machines/issues/29
Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1913548
Fixes https://bugzilla.redhat.com/show_bug.cgi?id=2030836

### CODE REVIEW

**Comment by KKoukiou** [comment]
This is now unblocked :raised_hands:

File: src/components/vm/consoles/vnc.jsx
Comment (KKoukiou) [inline src/components/vm/consoles/vnc.jsx:176]: Looks like scalling and clipping don't work well together? https://github.com/novnc/noVNC/blob/master/core/rfb.js#L344

**Comment by skobyda** [comment]
This Improved the scaling of VNC singificantly, but still breaks with higher resolution. 

Basically before this PR, there was offset between mouse position shown on screen and mouse position recorded by the system basically for every resolution, as shown [here](https://github.com/cockpit-project/cockpit-machines/pull/900#issue-1507838265), as shown here.

Now however the VNC scales properly for small resolution (e.g. 800x600), high resolution (e.g. 1080p), but scaling still breaks for ultra high resolution (3840x2160).

To show what I mean, here is a demo:

[presentation-(2).webm](https://user-images.githubusercontent.com/42733240/219353663-3a8e5949-ccfa-4dc9-b22f-03d88a0517b2.webm)

**Comment by skobyda** [comment]
But still, this improves usability of our VNC singificantly, so I'm very happy to merge this as it is.

**Comment by skobyda** [comment]
But hey, we are passing the "scaleViewport" and "resizeSession" flags to noVnc, so I guess that it doesn't scale and resize properly for ultra-high resolution is probably now a bug for [noVnc](https://github.com/novnc/noVNC)?

**Comment by KKoukiou** [comment]
> But hey, we are passing the "scaleViewport" and "resizeSession" flags to noVnc, so I guess that it doesn't scale and resize properly for ultra-high resolution is probably now a bug for [noVnc](https://github.com/novnc/noVNC)?

Yes, looks like that. I will check if exissts otherwise report. I will update pixel tests here, to get it in as partial fix.

**Comment by KKoukiou** [comment]
The debian installer is being heavily clipped. It does not seem to respond to the resize requests.

**Comment by garrett** [comment]
> The debian installer is being heavily clipped. It does not seem to respond to the resize requests.

Possibly silly suggestion: Is there a way to set a fallback/default resolution — like to 1024x768 — which is what would be used unless a size request works? If so, then Debian (and others) would then work, but those supporting size requests should have whatever resolution they need too.

**Comment by KKoukiou** [comment]
> > The debian installer is being heavily clipped. It does not seem to respond to the resize requests.
> 
> Possibly silly suggestion: Is there a way to set a fallback/default resolution — like to 1024x768 — which is what would be used unless a size request works? If so, then Debian (and others) would then work, but those supporting size requests should have whatever resolution they need too.

Unfortunately the API does not make this posisble.

**Comment by garrett** [comment]
Still a problem with Debian Testing, apparently:

image


Expanding works though:

image


Expanded, very narrow:

image


But if I resize so it's a little less vertical, it zooms in and crops:

image

**Comment by skobyda** [comment]
Ubuntu seems to work correctly:
Screenshot from 2023-02-21 11-52-05

The issue on ultra-high resolution for fedora persists:
Screenshot from 2023-02-21 11-51-05

**Comment by skobyda** [comment]
For some reason, debian-testing is not so bad for me. The bottom of the screen seems to be slightly cut off, but nothing as bad as seen in garret's screenshot:
Screenshot from 2023-02-21 11-57-05
Screenshot from 2023-02-21 11-56-56

**Comment by garrett** [comment]
> But still, this improves usability of our VNC singificantly, so I'm very happy to merge this as it is.

How ready is this? I know it's not perfect, but it's much better than we have, at least from when I looked at it. It would be nice to get it into this week's release.

**Comment by skobyda** [comment]
I'm personally happy with how it is now, as it improves stuff by a lot. Let's merge it! There are corner case issues, let's create a github issue to track it after this gets merged.

@KKoukiou  since I'm author of this PR, I can't approve it, so do you mind approving (if you are also happy with the current state) and merging. From my side it's LGTM

**Review by KKoukiou — APPROVED** [review]

**Pull request merged**
